ring/partition_ring_status.gohtml

```diff
@@ -50,12 +50,21 @@
                     <!-- Allow to force a state change -->
                     {{ if and (not .Corrupted) (ne (index $stateChanges .State) 0) }}
                         {{ $toState := index $stateChanges .State }}
-                        <form action="" method="POST" onsubmit="return confirm('Do you confirm you want to change the state to {{ $toState.CleanName }}?');">
+                        <form action="" method="POST">
                             <input type="hidden" name="csrf_token" value="$__CSRF_TOKEN_PLACEHOLDER__">
                             <input type="hidden" name="partition_id" value="{{ .ID }}">
                             <input type="hidden" name="partition_state" value="{{ $toState.String }}">
 
-                            <button name="action" value="change_state" type="submit">Change state to {{ $toState.CleanName }}</button>
+                            <button name="action" value="change_state" type="submit" {{ if .StateChangeLocked }}disabled title="State change is locked"{{ end }} onclick="return confirm('Do you confirm you want to change the state to {{ $toState.CleanName }}?');">Change state to {{ $toState.CleanName }}</button>
```

Comment:
Does the `onclick` here prevents the change if canceled? I'm wondering if the safest option would just be creating a `<form>` for each different action (in this case it would be 1 more form).